sdk/identity/azure-identity/inc/azure/identity/detail/token_cache.hpp

```diff
@@ -24,21 +26,15 @@ namespace Azure { namespace Identity { namespace _detail {
    * @brief Access token cache.
    *
    */
-  class TokenCache
-#if !defined(TESTING_BUILD)
-      final
-#endif
-  {
-#if !defined(TESTING_BUILD)
+  class TokenCache _azure_NON_FINAL_FOR_TESTS {
   private:
-#else
-  protected:
-#endif
```

Comment:
@LarryOsterman A good thing from this effort is we caught this `protected:` keyword with side effects, which was making unrelated structs like `CacheKey` etc. below protected, instead of private.

We can fix that now.